Source/WebKit/Shared/WebSQLiteDatabaseTracker.cpp

```diff
@@ -69,7 +69,12 @@ void WebSQLiteDatabaseTracker::willBeginFirstTransaction()
         return;
     }
 
-    setIsHoldingLockedFiles(true);
+    ensureOnMainRunLoop([weakThis = WeakPtr { *this }] {
+        if (RefPtr protectedThis = weakThis.get()) {
+            Locker locker { protectedThis->m_lock };
```

Comment:
Doesn't this deadlock when called on the main thread? The lock is already held starting on line 65.

Reply:
It doesn't happen in practice since we never do database work on the main thread but I've switch to a `RunLoop::main()->dispatch()` call to guarantee it. You're correct that we'd lock recursively otherwise.